swagger_parser/lib/src/parser/parser/open_api_parser.dart

```diff
@@ -639,8 +639,10 @@ class OpenApiParser {
         );
       }
 
-      if (value.containsKey(_propertiesConst)) {
-        findParametersAndImports(value);
+      if (value[_typeConst] == _objectConst) {
```

Comment:
Replace this two ifs nt one if (value.containsKey(_propertiesConst) && value[_typeConst] == _objectConst))

Reply:
It won't fix the bug. The goal is to always have an object class even without properties. With two ifs, it will become a typedef object which breaks the retrofit class generator.